libs/pages/services/src/lib/ui/page-settings-deployment-pipeline/page-settings-deployment-pipeline.tsx

```diff
@@ -1,80 +1,131 @@
+                      className="relative -top-[2px] ml-1 cursor-default text-xs text-neutral-350"
+                    />
+                  </div>
+                </Tooltip>
+              )}
+              <DropdownMenu.Root>
+                <DropdownMenu.Trigger asChild>
+                  <Button data-testid="btn-more-menu" color="neutral" variant="plain" className="ml-auto">
+                    <Icon iconName="ellipsis" />
+                  </Button>
+                </DropdownMenu.Trigger>
+                <DropdownMenu.Content>
+                  <DropdownMenu.Item
+                    data-testid="menuItem"
+                    icon={<Icon iconName="pen" />}
+                    onSelect={() =>
+                      openModal({
+                        content: <StageModalFeature onClose={closeModal} environmentId={environmentId} stage={stage} />,
+                      })
+                    }
+                  >
+                    Edit stage
+                  </DropdownMenu.Item>
+                  <DropdownMenu.Separator />
+                  <DropdownMenu.Item
+                    data-testid="menuItem"
+                    color="red"
+                    icon={<Icon iconName="trash" />}
+                    onSelect={() =>
+                      openModalConfirmation({
+                        title: 'Delete this stage',
+                        isDelete: true,
+                        name: stage.name,
+                        action: () => deleteEnvironmentDeploymentStage({ stageId: stage.id }),
+                      })
+                    }
+                  >
+                    Delete stage
+                  </DropdownMenu.Item>
+                </DropdownMenu.Content>
+              </DropdownMenu.Root>
+            </div>
+          ),
+          items:
+            stage.services?.map((s) => {
+              const service = services?.find(({ id }) => id === s.service_id)
+              const contentWithParams = service?.serviceType === 'DATABASE'
+              return {
+                ...s,
+                content: (
+                  <div className="flex flex-row items-center">
+                    {service && <ServiceAvatar className="mr-2" service={service} size="sm" border="solid" />}
+                    <div className={`font-medium text-neutral-400 ${contentWithParams ? 'text-xs' : 'text-ssm'}`}>
+                      <Truncate truncateLimit={contentWithParams ? 32 : 27} text={service?.name || ''} />
+                      {contentWithParams && (
+                        <div data-testid="draggable-item-subtitle" className="text-2xs font-normal">
+                          {upperCaseFirstLetter(service.type)} - {upperCaseFirstLetter(service.mode)}
+                        </div>
+                      )}
+                    </div>
+                  </div>
+                ),
+              }
+            }) ?? [],
+        } satisfies ColumnType
+      }),
+    [services, stages]
```

Comment:
Missing dependencies: `closeModal`, `deleteEnvironmentDeploymentStage`, `environmentId`, `openModal`, and `openModalConfirmation`

```suggestion
    [services, stages, closeModal, deleteEnvironmentDeploymentStage, environmentId, openModal, openModalConfirmation]
```

Reply:
Fixed https://github.com/Qovery/console/pull/1671/commits/d71ebda2c2f6f44a1ffcfabff1c180728877fbf0